### PR TITLE
Introduce tracing to logging of papiea components

### LINKED ISSUE #576: Add tracing to logs
The current goto in terms of tracing and distributed systems is OpenTracing (https://opentracing.io/) with Jaeger (https://www.jaegertracing.io/) as an implementation. We can plug it in and track:

Client -> Engine -> Provider -> Engine path
(Possibly via spec version) Client -> Engine -> Diff Resolver loop
Engine -> DB -> Engine
Besides these, we can plug trace id to the regular logger

### PR DESCRIPTION
**Pull request by Emulebest**
resolves #576

### CODE REVIEW

File: papiea-client/src/entity_client.ts [hunk 1]
```diff
@@ -65,43 +71,64 @@ function make_request<T = any, Y = AxiosPromise<T>>(f: (url: string, data?: any,
...
-    }, { headers: { "Authorization": `Bearer ${ s2skey }`, "Papiea-Version": `${ PAPIEA_VERSION }` } });
+    }, {headers});
+    span.finish()
     return watcher
 }
 
-async function get_entity(provider: string, kind: string, version: string, entity_reference: Entity_Reference, papiea_url: string, s2skey: string): Promise<Entity> {
+async function get_entity(provider: string, kind: string, version: string, entity_reference: Entity_Reference, papiea_url: string, s2skey: string, tracer: Tracer): Promise<Entity> {
+    const headers = getHeaders(s2skey)
+    const span = spanOperation("update_entity_client", {headers, tracer}, {entity_uuid: entity_reference.uuid})
```
Comment (shlomi-nx) [inline papiea-client/src/entity_client.ts:97]: Shouldn't this be "get_entity_client"?

File: papiea-engine/package.json [hunk 2]
```diff
@@ -14,8 +14,8 @@
     "build!": "npm run clean && npm run build",
     "start": "node build/main.js",
     "start_differ": "node build/diff_resolver_main.js",
-    "test": "jest --runInBand --coverage",
-    "test-ci": "jest --runInBand --coverage --ci --reporters=default --reporters=jest-junit",
+    "test": "jest --runInBand",
+    "test-ci": "jest --runInBand --ci --reporters=default --reporters=jest-junit",
```
Comment (shlomi-nx) [inline papiea-engine/package.json:18]: Why remove coverage?

File: papiea-engine/src/intentful_core/entity_creation_strategies/constructor_entity_creation_strategy.ts [hunk 3]
```diff
@@ -100,9 +104,12 @@ export class ConstructorEntityCreationStrategy extends EntityCreationStrategy {
                     this.validator.validate(input, Object.values(constructor.argument)[0], schemas,
                                             this.provider.allowExtraProps,
                                             Object.keys(constructor.argument)[0], "Constructor procedure")
+                    const span = spanOperation(`constructor`,
```
Comment (shlomi-nx) [inline papiea-engine/src/intentful_core/entity_creation_strategies/constructor_entity_creation_strategy.ts:107]: Perhaps we should call this "custom_constructor"?

**Review by shlomi-nx — COMMENTED** [review]
Looks good so far, minor comments.

File: papiea-engine/package.json (hunk 2 above)
Comment (Emulebest) [reply papiea-engine/package.json:18]: This introduces a critical bug in current versions of jest + one of deps in jaeger lib. Removed it for now (til the bug is fixed), as we are not using the coverage atm after all

File: papiea-engine/src/intentful_core/entity_creation_strategies/constructor_entity_creation_strategy.ts (hunk 3 above)
Comment (Emulebest) [reply papiea-engine/src/intentful_core/entity_creation_strategies/constructor_entity_creation_strategy.ts:107]: Good idea!

File: papiea-client/src/entity_client.ts (hunk 1 above)
Comment (Emulebest) [reply papiea-client/src/entity_client.ts:97]: Oops, seems like I missed it, thanks

File: papiea-client/src/entity_client.ts [hunk 4]
```diff
@@ -168,13 +203,17 @@ export interface ProviderClient {
...
-export function provider_client(papiea_url: string, provider: string, version: string, s2skey?: string, meta_extension?: (s2skey: string) => any): ProviderClient {
+export function provider_client(papiea_url: string, provider: string, version: string, s2skey?: string, tracer?: Tracer): ProviderClient {
+    const client_tracer = tracer ?? getTracer("papiea-client")
     const the_s2skey = s2skey ?? 'anonymous'
     return {
-        get_kind: (kind: string) => kind_client(papiea_url, provider, kind, version, the_s2skey, meta_extension),
-        invoke_procedure: (proc_name: string, input: any) => invoke_provider_procedure(provider, version, proc_name, input, papiea_url, the_s2skey)
+        get_kind: (kind: string) => kind_client(papiea_url, provider, kind, version, the_s2skey),
+        invoke_procedure: (proc_name: string, input: any) => invoke_provider_procedure(provider, version, proc_name, input, papiea_url, the_s2skey, client_tracer),
+        close: () => (client_tracer as any).close()
```
Comment (shlomi-nx) [inline papiea-client/src/entity_client.ts:216]: @Ntnx-Igor-Korotach Why was the `meta_extension` removed?

**Pull request merged**